src/gfx/reverse.cpp

```diff
@@ -254,71 +254,133 @@ void reverse() {
 	}
 
 	std::optional<DefaultInitVec<uint8_t>> attrmap;
+	uint16_t nbTilesInBank[] = {0, 0}; // Only used if there is an attrmap.
 	if (!options.attrmap.empty()) {
 		attrmap = readInto(options.attrmap);
-		if (attrmap->size() != nbTileInstances) {
+		if (attrmap->size() != mapSize) {
 			fatal(
 			    "Attribute map size (%zu tiles) doesn't match image's (%zu)",
 			    attrmap->size(),
-			    nbTileInstances
+			    mapSize
 			);
 		}
 
 		// Scan through the attributes for inconsistencies
 		// We do this now for two reasons:
 		// 1. Checking those during the main loop is harmful to optimization, and
 		// 2. It clutters the code more, and it's not in great shape to begin with
-		bool bad = false;
-		for (auto attr : *attrmap) {
+		for (size_t index = 0; index < mapSize; ++index) {
+			uint8_t attr = (*attrmap)[index];
+			size_t tx = index % width, ty = index / width;
+
 			if ((attr & 0b111) > palettes.size()) {
 				error(
-				    "Referencing palette %u, but there are only %zu!", attr & 0b111, palettes.size()
+				    "Attribute map references palette #%u at (%zu, %zu), but there are only %zu!",
+				    attr & 0b111,
+				    tx,
+				    ty,
+				    palettes.size()
 				);
-				bad = true;
 			}
-			if (attr & 0x08 && !tilemap) {
-				warning("Tile in bank 1 but no tilemap specified; ignoring the bank bit");
+
+			bool bank = attr & 0x08;
+
+			if (!tilemap) {
+				if (bank) {
+					warning(
+					    "Attribute map assigns tile at (%zu, %zu) to bank 1, but no tilemap "
+					    "specified; "
+					    "ignoring the bank bit",
+					    tx,
+					    ty
+					);
+				}
+			} else {
+				if (uint8_t tileOfs = (*tilemap)[index] - options.baseTileIDs[bank];
+				    tileOfs >= nbTilesInBank[bank]) {
+					nbTilesInBank[bank] = tileOfs + 1;
+				}
+			}
+		}
+
+		options.verbosePrint(
+		    Options::VERB_INTERM,
+		    "Nb tiles in bank {0: %" PRIu16 ", 1: %" PRIu16 "}\n",
+		    nbTilesInBank[0],
+		    nbTilesInBank[1]
+		);
+
+		for (int bank = 0; bank < 2; ++bank) {
```

Comment:
...ah, `NB_TILE_BANKS` is not defined, so we get build failures. I'll revert this back to `2`, it's quite obvious what the `2` is about, since the index variable is quite explicitly called `bank`.